### PR TITLE
added Pkg Filer

### PR DESCRIPTION
**Pull request by matmoeh**
moved the pfSense Package "Filer" to pfsense 2.3
"make package" and "pkg install" worked
Did not get rid of the default "Save"-Button under "Diagnostics" -> "Filer" -> "Files" tab

### CODE REVIEW

**Comment by pfSense-git-updates** [comment]
Before this pull request can be accepted you must first sign a CLA as described at https://www.pfsense.org/about-pfsense/#cla. Please read for more details.

**Comment by matmoeh** [comment]
Signed the CLA!

**Comment by doktornotor** [comment]
The XMLRPC stuff needs to get fixed to work on 2.4. Do not include xmlrpc.inc and see e.g. https://github.com/pfsense/FreeBSD-ports/blob/devel/www/pfSense-pkg-squid/files/usr/local/pkg/squid.inc#L2300 for example of required changes

**Comment by matmoeh** [comment]
Ok, did the xmlrpc 2.4 adjustments basing on squid.inc, but i can not test this!

File: sysutils/pfSense-pkg-filer/files/etc/inc/priv/filer.priv.inc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/files/etc/inc/priv/filer.priv.inc]: License has been moved to Apache on all pfSense packages on branch devel, please fix license text accordingly

File: sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.inc [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+/* ========================================================================== */
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.inc]: Fix license text

File: sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.inc [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+require_once("config.inc");
+require_once("globals.inc");
+require_once("notices.inc");
+require_once("util.inc");
+require_once("xmlrpc.inc");
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.inc]: xmlrpc.inc has been removed from 2.4, it should be required only in 2.3

File: sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.xml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+/* ====================================================================================== */
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.xml]: Fix license text

File: sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.xml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+	]]>
+	</copyright>
+	<name>filer</name>
+	<version>0.60.6</version>
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer.xml]: Version should be hardcoded here, replace it with %%PKGVERSION%% and add filer.xml to the list of files to replace this macro on Makefile

File: sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer_sync.xml [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+/* ====================================================================================== */
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer_sync.xml]: Fix license text

File: sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer_sync.xml [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+	]]>
+	</copyright>
+	<name>filersync</name>
+	<version>0.60.6</version>
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/files/usr/local/pkg/filer_sync.xml]: Version should be hardcoded here, replace it with %%PKGVERSION%% and add this file to the list in Makefile

File: sysutils/pfSense-pkg-filer/pkg-plist [hunk 8]
```diff
@@ -0,0 +1,6 @@
+pkg/filer.xml
+pkg/filer_sync.xml
+pkg/filer.inc
+/etc/inc/priv/filer.priv.inc
+%%DATADIR%%/info.xml
+@dir /etc/inc/priv
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-filer/pkg-plist]: It's missing '@dir /etc/inc' item here

**Comment by rbgarga** [comment]
@matmoeh package looks OK now. Could you please squash all these commits so we add the new port with a single commit? It would make it easier to backport to 2.3.3 after more testing

**Comment by matmoeh** [comment]
Im sorry, i dont know how to squash all the commits. You should be able to select squash as merge option?
https://github.com/blog/2141-squash-your-commits

**Comment by rbgarga** [comment]
@matmoeh unfortunately not since we merge into our internal repo first, and it syncs with github later. No problems, I'll apply it as a single patch keeping you as author and manual commit it

**Comment by doktornotor** [comment]
@rbgarga - What happened with this?

**Comment by matmoeh** [comment]
When will this be included in the current PFSense version? Waiting for this since nearly a year. Still using version 2.2.6, because i need this package. I also bought Gold Subscription....

**Comment by luckman212** [comment]
Is there a reason this package doesn't show up in 2.4? @rbgarga

**Comment by luckman212** [comment]
Thanks @jim-p !

**Pull request merged**